src/torrent/utils/thread_interrupt.cc

```diff
@@ -73,6 +70,8 @@ thread_interrupt::event_read() {
     throw internal_error("Invalid result reading from thread_interrupt socket.");
 
   instrumentation_update(INSTRUMENTATION_POLLING_INTERRUPT_READ_EVENT, 1);
+
+  m_poking = false;
```

Comment:
I didn't dig into exactly why, but moving this back down is indeed required for the tests to succeed.